### PR TITLE
Site Media: Improve media details screen

### PR DESCRIPTION
**Pull request by kean**
Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/21457.

I've discussed with Chris which changes to prioritize for the 23.7 release the it comes to the Media details view. This is the best bank for a buck:

- Allow swiping between media items left and right
- Update navigation bar icons
- Update table view style
- Use optimistic UI for savings metadata changes
- When you delete an item and there are more item in the media library, switch to the adjacent one
- Use “correct” multiplication sign in dimensions: ×
- Remove vertical scroll indicator

To test:

- Verify based on the "acceptance criteria",
- Verify that the metadata does get saved on the server and the when you updated the title, it updated the navigation bar title
- Verify the edge cases: deleting first and last item

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/926f57a6-bd07-410e-a270-bd59e6f22eb8

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -75,7 +75,7 @@
       },
       {
         "package": "swift-numerics",
-        "repositoryURL": "https://github.com/apple/swift-numerics",
+        "repositoryURL": "https://github.com/apple/swift-numerics.git",
```
Comment (momo-ozawa) [inline WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Did you mean to check in this change?

**Review by momo-ozawa — APPROVED** [review]

File: WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (kean) [reply WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Xcode 15 misbehaving. Fixed.
